hardware/arduino/avr/cores/arduino/Tone.cpp

```diff
@@ -456,19 +457,19 @@ void disableTimer(uint8_t _timer)
 
 #if defined(TIMSK3)
     case 3:
-      TIMSK3 = 0;
+      TIMSK3 &= ~(1 << OCIE3A);
```

Comment:
According to the ATmega32u4 datasheet, the Leonardo doesn't have a timer 5 and therefore does not have TIMSK5. Nevertheless, iom32u4.h defines TIMSK5, but without defining individual bits in it:

```
...
#define TIMSK4 _SFR_MEM8(0x72)
#define TOIE4 2
#define OCIE4B 5
#define OCIE4A 6
#define OCIE4D 7

#define TIMSK5 _SFR_MEM8(0x73)

#ifndef __ASSEMBLER__
#define ADC _SFR_MEM16(0x78)
#endif
#define ADCW _SFR_MEM16(0x78)
...
```

So the main question is why the CPU header defines TIMSK5 at all. But at least it explains the partly complicated #if defined(...) sequences in Tone.cpp, where the presence of all macros is carefully checked before actually doing something.

I've fixed this now by extending these preprocessor checks. The toneMelody example was tested with these changes on the following platforms:

* Compiles & runs on Leonardo
* Compiles & runs on "bare" ATmega8 (\w 16 MHz XTAL)
* Compiles on Uno
* Compiles on Yun
* Compiles on LilyPad
* Compiles on Mini (\w ATmega168)

I'll push the new fix in a minute.